### PR TITLE
Snowflake Integration update

### PR DESCRIPTION
**Pull request by TJaniF**
Issue: #1633 

- Changes the Integration to fit the tutorial format more closely.
- Added the SQL statements in a tutorial step so users don't have to copy them from GitHub (mainly because the data quality repo has much more info including on other tools than is needed for this tutorial, so I thought it might be a little overwhelming to navigate for someone just looking to learn how to integrate Snowflake). 
- Some minor updates to the rest of the content that I moved into a How it works section: new lineage graph, as well as shortened the chapter on available operators.

### CODE REVIEW

File: learn/airflow-snowflake.md [hunk 1]
```diff
@@ -5,219 +5,485 @@ id: airflow-snowflake
...
-    default_args={"conn_id": SNOWFLAKE_CONN_ID, "snowflake_conn_id": SNOWFLAKE_CONN_ID}
-) as dag:
+    ![Complex Snowflake DAG](/img/guides/snowflake_complex_dag.png)
 
-    create_forestfire_audit_table = SnowflakeOperator(
-        task_id="create_forestfire_audit_table",
-        sql="create_forestfire_table.sql",
-        params={"table_name": SNOWFLAKE_AUDIT_TABLE},
-    )
+    The DAG accomplishes the following steps:
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:431]:
```suggestion
    The DAG completes the following steps:
```

File: learn/airflow-snowflake.md [hunk 2]
```diff
@@ -5,219 +5,485 @@ id: airflow-snowflake
...
 
-    create_forestfire_production_table = SnowflakeOperator(
-        task_id="create_forestfire_production_table",
-        sql="create_forestfire_table.sql",
-        params={"table_name": SNOWFLAKE_FORESTFIRE_TABLE}
-    )
+    - Creating three tables simultaneously using the [SnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/snowflakeoperator).
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:433]:
```suggestion
    - Creates three tables simultaneously using the [SnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/snowflakeoperator). 
```

File: learn/airflow-snowflake.md [hunk 3]
```diff
@@ -5,219 +5,485 @@ id: airflow-snowflake
...
-    )
+    The DAG accomplishes the following steps:
 
-    create_forestfire_production_table = SnowflakeOperator(
-        task_id="create_forestfire_production_table",
-        sql="create_forestfire_table.sql",
-        params={"table_name": SNOWFLAKE_FORESTFIRE_TABLE}
-    )
+    - Creating three tables simultaneously using the [SnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/snowflakeoperator). 
+    - Loading data into two of the tables that were created using the [SnowflakeOperatorAsync](https://registry.astronomer.io/providers/astronomer-providers/modules/snowflakeoperatorasync) a deferrable version of the SnowflakeOperator. In order to be able to use deferrable operators you need to have a triggerer component running in your Airflow environment, which is configured automatically if you are using the Astro CLI. Learn more about deferrable operators in our [Deferrable operators guide](deferrable-operators.md).
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:434]:
```suggestion
    - Loads data into two of the tables that were created using the [SnowflakeOperatorAsync](https://registry.astronomer.io/providers/astronomer-providers/modules/snowflakeoperatorasync) a deferrable version of the SnowflakeOperator. In order to be able to use deferrable operators you need to have a triggerer component running in your Airflow environment, which is configured automatically if you are using the Astro CLI. Learn more about deferrable operators in our [Deferrable operators guide](deferrable-operators.md).
```
Can move the deferrable notes into the separate step below.

File: learn/airflow-snowflake.md [hunk 4]
```diff
@@ -5,219 +5,485 @@ id: airflow-snowflake
...
+    - Running data quality checks on the data to ensure that no erroneous data is moved to production. These checks are structured with [task groups](task-groups.md) that include column checks using the [SQLColumnCheckOperator](https://registry.astronomer.io/providers/common-sql/modules/sqlcolumncheckoperator) and table checks using the [SQLTableCheckOperator](https://registry.astronomer.io/providers/common-sql/modules/sqltablecheckoperator).
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:435]:
```suggestion
    - Runs data quality checks on the data to ensure that no erroneous data is moved to production. These checks are structured with [task groups](task-groups.md) that include column checks using the [SQLColumnCheckOperator](https://registry.astronomer.io/providers/common-sql/modules/sqlcolumncheckoperator) and table checks using the [SQLTableCheckOperator](https://registry.astronomer.io/providers/common-sql/modules/sqltablecheckoperator).
```

File: learn/airflow-snowflake.md [hunk 5]
```diff
@@ -5,219 +5,485 @@ id: airflow-snowflake
...
+    - Copying data into the production table using the SnowflakeOperatorAsync.
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:436]:
```suggestion
    - Copies data into the production table using the SnowflakeOperatorAsync.
```

File: learn/airflow-snowflake.md [hunk 6]
```diff
@@ -5,219 +5,485 @@ id: airflow-snowflake
...
+    - Deleting the tables to clean up the example.
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:437]:
```suggestion
    - Deletes the tables to clean up the example.
```

**Review by kentdanas — CHANGES_REQUESTED** [review]
Looks great! Requesting changes to add a few additional steps in the tutorial part to highlight data quality and async functionality a bit more. Otherwise just minor edits.

**Review by TJaniF — COMMENTED** [review]
Thank you! @kentdanas :) 
I added the additional steps as we discussed. @jwitz I think this one is ready for your review :)

File: learn/airflow-snowflake.md [hunk 7]
```diff
@@ -5,219 +5,539 @@ id: airflow-snowflake
...
-            task_id="column_checks",
-            table=SNOWFLAKE_AUDIT_TABLE,
-            column_mapping={"id": {"null_check": {"equal_to": 0}}}
+The `complex_snowflake_example` DAG runs several queries against the same Snowflake database in parallel. While some queries like the ones creating tables run quickly, larger transformation or loading queries might take some time to run. This is a great use case for the deferrable version of the SnowflakeOperator, the [SnowflakeOperatorAsync](https://registry.astronomer.io/providers/astronomer-providers/modules/snowflakeoperatorasync). Deferrable operators use the triggerer component in your Airflow environment, which is configured automatically if you are using the Astro CLI, in order to release their worker slot while they wait to be able to execute a task. This allows you to use your Airflow resources much more efficiently in production. Learn more about deferrable operators in our [Deferrable operators guide](deferrable-operators.md).
+
+Using deferrable operators from the Astronomer providers package is easy, you simply have to switch out the operator class. All parameters stay the same.
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:465]: Link to the astro provider repo here?

File: learn/airflow-snowflake.md [hunk 8]
```diff
@@ -5,219 +5,539 @@ id: airflow-snowflake
...
+
+1. Add the following import statement to your DAG importing the SnowflakeOperatorAsync:
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:467]:
```suggestion
1. Add the following statement to your DAG to import the SnowflakeOperatorAsync:
```

File: learn/airflow-snowflake.md [hunk 9]
```diff
@@ -5,219 +5,539 @@ id: airflow-snowflake
...
+    from astronomer.providers.snowflake.operators.snowflake import (
+            SnowflakeOperatorAsync
         )
+    ```
 
-        table_checks = SQLTableCheckOperator(
-            task_id="table_checks",
-            table=SNOWFLAKE_AUDIT_TABLE,
-            checks={"row_count_check": {"check_statement": "COUNT(*) = 9"}}
+2. Switch out the operator used in the `load_forestfire_data` and `load_cost_data` task by replacing `SnowflakeOperator` with `SnowflakeOperatorAsync`:
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:475]:
```suggestion
2. Switch out the operators used in the `load_forestfire_data` and `load_cost_data` tasks by replacing `SnowflakeOperator` with `SnowflakeOperatorAsync`:
```

File: learn/airflow-snowflake.md [hunk 10]
```diff
@@ -5,219 +5,539 @@ id: airflow-snowflake
...
         )
+    ```
 
-    delete_audit_table = SnowflakeOperator(
-        task_id="delete_audit_table",
-        sql="delete_forestfire_table.sql",
-        params={"table_name": f"{SNOWFLAKE_FORESTFIRE_TABLE}_AUDIT"},
-        trigger_rule="all_success"
-    )
+3. Run your DAG and observe how the two load tasks go into a deferred state (purple border) before being executed.
```
Comment (kentdanas) [inline learn/airflow-snowflake.md:491]:
```suggestion
3. Run your DAG and observe how the two load tasks go into a deferred state (purple border) before being completed.
```

File: learn/airflow-snowflake.md [hunk 11]
```diff
@@ -5,219 +5,539 @@ id: airflow-snowflake
...
-        end
-    )
-```
+     ![Load tasks deferred state](/img/guides/snowflake_example_deferred_state.png)
 
-![Snowflake DAG Graph](/img/guides/snowflake_dag_graph.png)
+## How it works
 
-To run this DAG, you need an Airflow connection to your Snowflake instance. This DAG uses a connection named `snowflake_default`. Your connection should have a type of `Snowflake` include the following information:
+This section provides additional information on orchestrating actions in Snowflake with Airflow.
```
Comment (jwitz) [inline learn/airflow-snowflake.md:497]: I feel like "How it works" is a bit of a stretch here based on the content, as the above DAG is not referenced. I'm happy to not include a "How it works" section if the DAG is straightforward enough

**Review by jwitz — CHANGES_REQUESTED** [review]
I like the new structure of this tutorial @TJaniF ! Requesting changes because I think we can provide more information about the "why" for this DAG. Why did we structure it the way we did, why did we use certain operators, and what's the best way to organize your Astro project? Left some comments elaborating on this throughout.

**Review by TJaniF — COMMENTED** [review]
Thank you @jwitz and @kentdanas ! I accepted almost all suggestions and added a few comments. :)

File: learn/airflow-snowflake.md (hunk 11 above)
Comment (kentdanas) [reply learn/airflow-snowflake.md:492]: We still want to include the additional information here - it was part of the original guide intentionally to highlight the benefits of using Snowflake with Airflow, but I don't think it really makes sense to pull it out as a separate concept guide given how specific it is. We don't need to call it "How it works" though - I'd be fine with "Additional information" or maybe something like "More on the Airflow Snowflake integration"...

File: learn/airflow-snowflake.md [hunk 12]
```diff
@@ -5,219 +5,539 @@ id: airflow-snowflake
...
 
-This guide covers the following topics:
+This tutorial covers an example of orchestrating a complex pattern in Snowflake with Airflow, including:
+
+- Creating tables.
+- Loading data into Snowflake.
+- Running any transformation on data in Snowflake using Airflow operators. 
+- Running data quality checks on data in Snowflake. 
+
+Additionally, the [How it works](#how-it-works) sections offers further information on:
```
[the remark this answers is not in the document]
Comment (jwitz) [reply learn/airflow-snowflake.md:17]: Good call! Missed that, updated the suggestion

File: learn/airflow-snowflake.md [hunk 13]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
-with DAG(
-    "snowflake_write_audit_publish",
-    description="Example DAG showcasing loading and data quality checking with Snowflake.",
-    start_date=datetime(2021, 1, 1),
-    schedule_interval=None,
-    template_searchpath="/usr/local/airflow/include/sql/snowflake_examples/",
-    catchup=False,
-    default_args={"conn_id": SNOWFLAKE_CONN_ID, "snowflake_conn_id": SNOWFLAKE_CONN_ID}
-) as dag:
+    The `chain()` method at the end of the DAG sets the dependencies. This method is commonly used over bitshift operators (`>>`) to make the dependency related code of DAGs with a lot of tasks easier to read.
```
Comment (jwitz) [inline learn/airflow-snowflake.md:446]:
```suggestion
    The `chain()` method at the end of the DAG sets the dependencies. This method is commonly used over bitshift operators (`>>`) to make it easier to read dependencies between many tasks.
```

File: learn/airflow-snowflake.md [hunk 14]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
 
-    with TaskGroup(group_id="quality_checks") as quality_check_group:
+## Step 6: Use deferrable operators
 
-        column_checks = SQLColumnCheckOperator(
-            task_id="column_checks",
-            table=SNOWFLAKE_AUDIT_TABLE,
-            column_mapping={"id": {"null_check": {"equal_to": 0}}}
+The `complex_snowflake_example` DAG runs several queries against the same Snowflake database in parallel. While some queries, like the ones creating tables, run quickly, larger transformation or loading queries might take longer to complete. These queries are a great use case for the deferrable version of the SnowflakeOperator, the [SnowflakeOperatorAsync](https://registry.astronomer.io/providers/astronomer-providers/modules/snowflakeoperatorasync). Deferrable operators use the triggerer component in your Airflow environment, which is configured automatically with the Astro CLI, to release their worker slot while they wait to for the task to be completed. This allows you to use your Airflow resources much more efficiently in production. Learn more about deferrable operators in our [Deferrable operators guide](deferrable-operators.md).
```
Comment (jwitz) [inline learn/airflow-snowflake.md:458]:
```suggestion
The `complex_snowflake_example` DAG runs several queries against the same Snowflake database in parallel. While some queries, like the ones creating tables, run quickly, larger transformation or loading queries might take longer to complete. These queries are a great use case for the deferrable version of the SnowflakeOperator, the [SnowflakeOperatorAsync](https://registry.astronomer.io/providers/astronomer-providers/modules/snowflakeoperatorasync). Deferrable operators use the triggerer component in your Airflow environment, which is configured automatically with the Astro CLI, to release their worker slot while they wait for the task to be completed. This allows you to use your Airflow resources much more efficiently in production. Learn more about deferrable operators in our [Deferrable operators guide](deferrable-operators.md).
```

**Review by jwitz — APPROVED** [review]
LGTM! Nice edits

File: learn/airflow-snowflake.md [hunk 15]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
 
-This guide covers the following topics:
+This tutorial covers an example of orchestrating complex Snowflake operations with Airflow, including:
+
+- Creating tables.
+- Loading data into Snowflake.
+- Running transformations on data in Snowflake using Airflow operators. 
+- Running data quality checks on data in Snowflake. 
+
+Additionally, [How it works](#how-it-works) offers further information on:
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:17]:
```suggestion
Additionally, [More on the Airflow Snowflake integration](#more-on-the-airflow-snowflake-integration) offers further information on:
```

File: learn/airflow-snowflake.md [hunk 16]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
+## Step 2: Configure a Snowflake connection
+
+1. In the Airflow UI, go to **Admin** -> **Connections** and click **+**. 
+
+2. Create a new connection named `snowflake_default` and choose the `snowflake` connection type. Enter the following information:
+
+    - [Schema](https://docs.snowflake.com/en/sql-reference/sql/create-schema.html): Your Snowflake schema. 
+    - Login: Your Snowflake login username.
+    - Password: Your Snowflake password.
+    - Account: Your snowflake account in the format `xz12345`.
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:68]:
```suggestion
    - Account: Your Snowflake account in the format `xz12345`.
```

File: learn/airflow-snowflake.md [hunk 17]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
+
+    ```sh
+    $ astro dev start
+    ```
+
+## Step 2: Configure a Snowflake connection
+
+1. In the Airflow UI, go to **Admin** -> **Connections** and click **+**. 
+
+2. Create a new connection named `snowflake_default` and choose the `snowflake` connection type. Enter the following information:
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:63]:
```suggestion
2. Create a new connection named `snowflake_default` and choose the `Snowflake` connection type. Enter the following information:
```

File: learn/airflow-snowflake.md [hunk 18]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
+## Step 2: Configure a Snowflake connection
+
+1. In the Airflow UI, go to **Admin** -> **Connections** and click **+**. 
+
+2. Create a new connection named `snowflake_default` and choose the `Snowflake` connection type. Enter the following information:
+
+    - [Schema](https://docs.snowflake.com/en/sql-reference/sql/create-schema.html): Your Snowflake schema. 
+    - Login: Your Snowflake login username.
+    - Password: Your Snowflake password.
+    - Account: Your Snowflake account in the format `xz12345`.
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:68]:
```suggestion
    - Account: Your Snowflake account in the format `xy12345`.
```

File: learn/airflow-snowflake.md [hunk 19]
```diff
@@ -5,219 +5,534 @@ id: airflow-snowflake
...
+            and data quality checking with multiple datasets in Snowflake.
+        """,
+        doc_md=__doc__,
+        start_date=datetime(2022, 12, 1),
+        schedule_interval=None,
+        # defining the directory where SQL templates are stored
+        template_searchpath="/usr/local/airflow/include/sql/",
+        catchup=False
+    ) as dag:
+        """
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:242]:
```suggestion
    ) as dag:
    
        """
```

Comment (TJaniF) [inline learn/airflow-snowflake.md:242]:
```suggestion
    with DAG(
        "complex_snowflake_example",
        description="""
            Example DAG showcasing loading, transforming, 
            and data quality checking with multiple datasets in Snowflake.
        """,
        doc_md=__doc__,
        start_date=datetime(2022, 12, 1),
        schedule=None,
        # defining the directory where SQL templates are stored
        template_searchpath="/usr/local/airflow/include/sql/",
        catchup=False
    ) as dag:
    
        """
```

File: learn/airflow-snowflake.md [hunk 20]
```diff
@@ -5,219 +5,535 @@ id: airflow-snowflake
...
+
+Using deferrable operators from the [Astronomer providers package](https://registry.astronomer.io/providers/astronomer-providers) is easy, you simply have to switch out the operator class. All parameters stay the same.
+
+1. Add the following statement to your DAG to import the SnowflakeOperatorAsync:
+
+    ```python
+    from astronomer.providers.snowflake.operators.snowflake import (
+            SnowflakeOperatorAsync
         )
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:468]:
```suggestion
    from astronomer.providers.snowflake.operators.snowflake import (
        SnowflakeOperatorAsync
    )
```

File: learn/airflow-snowflake.md [hunk 21]
```diff
@@ -5,219 +5,535 @@ id: airflow-snowflake
...
-            table=SNOWFLAKE_AUDIT_TABLE,
-            checks={"row_count_check": {"check_statement": "COUNT(*) = 9"}}
+2. Switch out the operators used in the `load_forestfire_data` and `load_cost_data` tasks by replacing `SnowflakeOperator` with `SnowflakeOperatorAsync`:
+
+    ```python
+    load_forestfire_data = SnowflakeOperatorAsync( # changed operator name
+            task_id="load_forestfire_data",
+            sql=sql_stmts.load_forestfire_data,
+            params={"table_name": SNOWFLAKE_FORESTFIRE_TABLE}
         )
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:478]:
```suggestion
    load_forestfire_data = SnowflakeOperatorAsync(  # changed operator name
        task_id="load_forestfire_data",
        sql=sql_stmts.load_forestfire_data,
        params={"table_name": SNOWFLAKE_FORESTFIRE_TABLE}
    )
```

File: learn/airflow-snowflake.md [hunk 22]
```diff
@@ -5,219 +5,535 @@ id: airflow-snowflake
...
-                "audit_table_name": f"{SNOWFLAKE_FORESTFIRE_TABLE}_AUDIT",
-                "table_schema": table_props,
-                "col_string": col_string,
-            },
-            trigger_rule="all_success"
+    load_cost_data = SnowflakeOperatorAsync( # changed operator name
+            task_id="load_cost_data",
+            sql=sql_stmts.load_cost_data,
+            params={"table_name": SNOWFLAKE_COST_TABLE}
         )
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:484]:
```suggestion
    load_cost_data = SnowflakeOperatorAsync( # changed operator name
        task_id="load_cost_data",
        sql=sql_stmts.load_cost_data,
        params={"table_name": SNOWFLAKE_COST_TABLE}
    )
```

File: learn/airflow-snowflake.md [hunk 23]
```diff
@@ -5,219 +5,535 @@ id: airflow-snowflake
...
+        task_id="load_forestfire_data",
+        sql=sql_stmts.load_forestfire_data,
         params={"table_name": SNOWFLAKE_FORESTFIRE_TABLE}
     )
 
-    load_data = SnowflakeOperatorAsync(
-        task_id="insert_query",
-        sql="load_snowflake_forestfire_data.sql",
-        params={"table_name": SNOWFLAKE_AUDIT_TABLE}
+    load_cost_data = SnowflakeOperatorAsync( # changed operator name
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:480]:
```suggestion
    load_cost_data = SnowflakeOperatorAsync(  # changed operator name
```

File: learn/airflow-snowflake.md [hunk 24]
```diff
@@ -5,219 +5,535 @@ id: airflow-snowflake
...
-    )
-```
+Several open source packages contain operators used to orchestrate Snowflake in Airflow. If you are using the Astro CLI these packages will all be pre-installed in your project.
 
-![Snowflake DAG Graph](/img/guides/snowflake_dag_graph.png)
+The [Snowflake provider package](https://registry.astronomer.io/providers/snowflake) contains:
 
-To run this DAG, you need an Airflow connection to your Snowflake instance. This DAG uses a connection named `snowflake_default`. Your connection should have a type of `Snowflake` include the following information:
+- [SnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/snowflakeoperator): Executes any SQL query in Snowflake.
+- [S3ToSnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/s3tosnowflakeoperator): Executes an COPY command to load files from s3 to Snowflake.
```
Comment (TJaniF) [inline learn/airflow-snowflake.md:502]:
```suggestion
- [S3ToSnowflakeOperator](https://registry.astronomer.io/providers/snowflake/modules/s3tosnowflakeoperator): Executes a COPY command to load files from s3 to Snowflake.
```

**Pull request merged**